hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -312,8 +332,11 @@ public HoodieInstant transitionCompactionInflightToComplete(HoodieInstant inflig
   }
 
   private void createFileInAuxiliaryFolder(HoodieInstant instant, Option<byte[]> data) {
-    Path fullPath = new Path(metaClient.getMetaAuxiliaryPath(), instant.getFileName());
-    createFileInPath(fullPath, data);
+    if (metaClient.getMetadataVersion().isNullVersion()) {
+      // No need for auxiliary folder for newer versions
```

Comment:
add more context. it's not need only under a scenario right

Reply:
Done.